AmmoDriver.md

```diff
@@ -0,0 +1,339 @@
+
+## Using the Ammo.js API
+
+The Ammo.js API lacks any usage documentation. Instead, it is recommended to read the [Bullet 2.83 documentation](https://github.com/bulletphysics/bullet3/tree/master/docs), the [Bullet forums](https://pybullet.org/Bullet/phpBB3/) and the [Emscripten WebIDL Binder documentation](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/WebIDL-Binder.html). Note that the linked Bullet documentation is for Bullet 2.83, where as Ammo.js is using 2.82, so some features described in the documentation may not be available.
+
+Some things to note:
+
+- Not all classes and properties in Bullet are available. Each class and property has to be 'exposed' via a definition in [ammo.idl](https://github.com/MozillaReality/ammo.js/blob/hubs/master/ammo.idl).
+- There is no automatic garbage collection for instantiated Bullet objects. Any time you use the `new` keyword for a Bullet class you must also at some point (when you are done with the object) release the memory by calling `Ammo.destroy`.
+
+```js
+const vector3 = new Ammo.btVector3();
+... do stuff
+Ammo.destroy(vector3);
+```
+
+- Exposed properties on classes can be accessed via specially generated `get_<property>()` and `set_<property>()` functions. E.g. `rayResultCallback.get_m_collisionObject();`
+- Sometimes when calling certain functions you will receive a pointer object instead of an instance of the class you are epecting. Use `Ammo.wrapPointer` to "wrap" the pointer in the class you expected. E.g. `Ammo.wrapPointer(ptr, Ammo.btRigidBody);`
+- Conversely, sometimes you need the pointer of an object. Use `Ammo.getPointer`. E.g. `const ptr = Ammo.getPointer(object);`.
+
+In A-Frame, each entity's `btRigidBody` instance is exposed on the `el.body` property. To apply a quick push to an object, you might do the following:
+
+```html
+<a-scene>
+  <a-entity id="nyan" dynamic-body="shape: hull" obj-model="obj: url(nyan-cat.obj)"></a-entity>
+  <a-plane static-body></a-plane>
+</a-scene>
+```
+
+```javascript
+var el = sceneEl.querySelector('#nyan');
+const force = new Ammo.btVector3(0, 1, -0);
+const pos = new Ammo.btVector3(el.object3D.position.x, el.object3D.position.y, el.object3D.position.z);
+el.body.applyForce(force, pos);
+Ammo.destroy(force);
+Ammo.destroy(pos);
+```
+
+## Events
+
+| event         | description                                                                                         |
+| ------------- | --------------------------------------------------------------------------------------------------- |
+| `body-loaded` | Fired when physics body (`el.body`) has been created.                                               |
+| `collide`     | Fired when two bodies collide. `emitCollisionEvents: true` must be set on the `ammo-body`.          |
+| `collide-end` | Fired when two bodies stop colliding. `emitCollisionEvents: true` must be set on the `ammo-body`.   |
```

Comment:
These don't match the event names mentioned above – looks like this version is correct?